pkg/jobs/jobs.go

```diff
@@ -32,6 +33,33 @@ import (
 	"github.com/cockroachdb/redact"
 )
 
+// jobDumpTraceMode is the type that represents the mode in which a traceable
+// job will dump a trace zip.
+type jobDumpTraceMode int64
+
+const (
+	// A Traceable job will not dump a trace zip.
+	noDump jobDumpTraceMode = iota
+	// A Traceable job will dump a trace zip on failure.
+	dumpOnFail
+	// A Traceable job will dump a trace zip in any of paused, canceled, failed,
+	// succeeded states.
+	alwaysDump
+)
+
+var traceableJobDumpTraceMode = settings.RegisterEnumSetting(
+	"jobs.trace.force_dump_mode",
+	"determines the state in which all traceable jobs will dump their cluster wide, inflight, "+
+		"trace recordings. Traces may be dumped never, on fail, "+
+		"or in any other state i.e paused, canceled, failed, succeeded.",
```

Comment:
What do you think about phrasing this as something like "on any status change". I think phrasing it as dumping "always" may be a bit confusing? (ie What does always mean, is it a continuous dump? Does it dump periodically? etc)

Reply:
done, changed the mode from `always` to `onStatusChanged`.